src/Symfony/Component/Console/Descriptor/ApplicationDescription.php

```diff
@@ -134,15 +134,17 @@ private function inspectApplication()
     private function sortCommands(array $commands)
     {
         $namespacedCommands = array();
+        $globalCommands = array();
         foreach ($commands as $name => $command) {
             $key = $this->application->extractNamespace($name, 1);
             if (!$key) {
-                $key = '_global';
+                $globalCommands['_global'][$name] = $command;
```

Comment:
hmm, actually no. The switch from array_merge to array_shift won't work because the code expects to have a _global key for global commands. But the global key could be handle at the time of merging, keeping ``$globalCommands`` as the list of global commands (which is not the case currently)